libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -0,0 +1,337 @@
+#include <boost/algorithm/string/join.hpp>
+#include <graphene/chain/balance_object.hpp>
+
+
+namespace graphene { namespace es_objects {
+
+namespace detail
+{
+
+class es_objects_plugin_impl
+{
+   public:
+      es_objects_plugin_impl(es_objects_plugin& _plugin)
+         : _self( _plugin )
+      {  curl = curl_easy_init(); }
+      virtual ~es_objects_plugin_impl();
+
+      void updateDatabase( const vector<object_id_type>& ids , bool isNew);
+
+      graphene::chain::database& database()
+      {
+         return _self.database();
+      }
+
+      es_objects_plugin& _self;
+      std::string _es_objects_elasticsearch_url = "http://localhost:9200/";
+      uint32_t _es_objects_bulk_replay = 1000;
+      uint32_t _es_objects_bulk_sync = 2;
+      bool _es_objects_proposals = true;
+      bool _es_objects_accounts = true;
+      bool _es_objects_assets = true;
+      bool _es_objects_balances = true;
+      bool _es_objects_logs = true;
+      CURL *curl; // curl handler
+      vector <string> bulk;
+   private:
+      void PrepareProposal(const proposal_object* proposal_object);
+      void PrepareAccount(const account_object* account_object);
+      void PrepareAsset(const asset_object* asset_object);
+      void PrepareBalance(const balance_object* balance_object);
+      void SendBulk();
+      void createBulk(std::string type, std::string data, std::string id);
+};
+
+void es_objects_plugin_impl::updateDatabase( const vector<object_id_type>& ids , bool isNew)
+{
+
+   graphene::chain::database &db = database();
+
+   auto block_time = db.head_block_time();
+
+   // check if we are in replay or in sync and change number of bulk documents accordingly
+   uint32_t limit_documents = 0;
+   if((fc::time_point::now() - block_time) < fc::seconds(30))
+      limit_documents = _es_objects_bulk_sync;
+   else
+      limit_documents = _es_objects_bulk_replay;
+
+   if (curl && bulk.size() >= limit_documents) { // we are in bulk time, ready to add data to elasticsearech
+      SendBulk();
+   }
+
+   for(auto const& value: ids) {
+      if(value.space() == 1 && value.type() == 10 && _es_objects_proposals) {
+         auto obj = db.find_object(value);
+         auto p = static_cast<const proposal_object*>(obj);
+         PrepareProposal(p);
+      }
+      else if(value.space() == 1 && value.type() == 2 && _es_objects_accounts) {
+         auto obj = db.find_object(value);
+         auto a = static_cast<const account_object*>(obj);
+         PrepareAccount(a);
+      }
+      else if(value.space() == 1 && value.type() == 3 && _es_objects_assets) {
+         auto obj = db.find_object(value);
+         auto a = static_cast<const asset_object*>(obj);
+         PrepareAsset(a);
+      }
+      else if(value.space() == 1 && value.type() == 15 && _es_objects_balances) {
+         auto obj = db.find_object(value);
+         auto b = static_cast<const balance_object*>(obj);
+         PrepareBalance(b);
+      }
```

Comment:
It's best if we can also store account statistics, dynamic asset data, bitasset data and etc along with the main objects.